pkg/auth/token.go

```diff
@@ -78,6 +79,9 @@ type TokenValidatorConfig struct {
 
 	// AllowPrivateIP allows JWKS/OIDC endpoints on private IP addresses
 	AllowPrivateIP bool
+
+	// ResourceURL is the explicit resource URL for OAuth discovery (RFC 9728)
+	ResourceURL string
```

Comment:
I don't love that I added the ResourceURL to the TokenValidatorConfig - would it be better to just create a separate ProtectedResourceConfig which would in the most common case be initialized the same as TokenValidatorConfig? The counter-argument was that I didn't want to add too many new flags to thv